firmware/application/apps/ui_freqman.cpp

```diff
@@ -308,31 +336,44 @@ FrequencyManagerView::FrequencyManagerView(
     };
 
     button_add_category.on_select = [this]() {
+        // this one will only directly add to /USR/FREQMAN
         on_add_category();
     };
 
-    button_del_category.on_select = [this]() {
-        on_del_category();
+    button_del_category.on_select = [this, &nav]() {
+        if (!forbid_delete_system_item_helper(nav)) {
+            on_del_category();
+        }
     };
 
-    button_edit_entry.on_select = [this](Button&) {
-        on_edit_entry();
+    button_edit_entry.on_select = [this, &nav](Button&) {
+        if (!forbid_delete_system_item_helper(nav)) {
+            on_edit_entry();
+        }
     };
 
-    button_edit_freq.on_select = [this](Button&) {
-        on_edit_freq();
+    button_edit_freq.on_select = [this, &nav](Button&) {
+        if (!forbid_delete_system_item_helper(nav)) {
```

Comment:
If someone wants to edit a system FREQMAN file, why don't we let them edit it and just save the modified file in the /USR/FREQMAN folder?

Reply:
Yes, that’s what I think and I write. You remind me that this helper should rename to others since I don’t only use this as delete.

Reply:
oh i read what you said wrong

Reply:
sorry for my English, you mean save a copy? sorry but i don't think i can do it, since it will do too much duplicate

Reply:
And about the implementation part:
what to do?
- save change to buffer and when exit app, write to file? not sure if memory is enough
- Or copy a duplicate and jump to it to let them edit?

Reply:
well ill try the second way

Reply:
ah it would be too much work to get file path from index. I think I'll leave this as a TODO. This PR already big enough and hard to read if i add more stuff.
I'm sorry

Reply:
The second way was my preference (make a copy and open the copy), but I understand if this has to wait for another PR.

Reply:
Maybe we don't need to make a copy of the file automatically.  Maybe we could prompt the user in the "forbid" code and give them a choice whether to make a copy.  The complication here is that there might already be a file of the same name in the USR folder.

Reply:
I think this will come to next PR, since current logic also make sense (but i admit not perfect)

> but I understand if this has to wait for another PR.

Sure, thanks!